File: src/main/java/jp/dataforms/fw/app/menu/page/SiteMapForm.java
package jp.dataforms.fw.app.menu.page;

import jp.dataforms.fw.menu.Menu;

/**
 * サイドメニューフォームクラス。
 *
 */
public class SiteMapForm extends MenuForm {
	/**
	 * コンストラクタ。
	 */
	public SiteMapForm() {
		super(null);
	}

	@Override
	protected Menu newMenuComponent() {
		Menu menu = new Menu();
		return menu;
	}
}

File: src/main/java/META-INF/resources/frame/flex/SiteMapForm.js
/**
 * メニューフォーム.
 */

'use strict';

/**
 * メニューフォーム.
 */
class SiteMapForm extends MenuForm {
	/**
	 * ページの各エレメントとの対応付け.
	 */
	attach() {
		super.attach();
		this.menu = this.newInstance(this.menu);
		this.menu.init();
		this.menu.attach();
	}
}

